app/experimenter/templates/experiments/edit_objectives.html

```diff
@@ -12,6 +12,10 @@
   {% include "experiments/field_inline.html" with field=form.objectives marked_required=True %}
   {% include "experiments/field_inline.html" with field=form.analysis marked_required=True %}
 
+  {% if not object.is_rollout or object.is_generic_experiment %}
```

Comment:
Put this on a model method like `should_have_total_enrolled` and check the types in there